### PR TITLE
Upgrade wildfly-common to 1.5 and xnio to 3.7 and restEasy to 4.2.0.Final

### PR DESCRIPTION
**Pull request by ge0ffrey**
Fixes this mess:
https://stackoverflow.com/questions/58014229/mvn-quarkusdev-throws-noclassdeffounderror-could-not-initialize-class-org-j

### CODE REVIEW

**Comment by ge0ffrey** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
@ge0ffrey why do we here care about quarkus compatibility?

**Comment by ge0ffrey** [comment]
Because the optaplanner repo hasn't forked.

**Comment by ge0ffrey** [comment]
Jenkins execute full downstream build

File: pom.xml [hunk 1]
```diff
@@ -321,7 +321,7 @@
     <version.org.wildfly>14.0.1.Final</version.org.wildfly>
     <version.org.wildfly.arquillian>2.1.1.Final</version.org.wildfly.arquillian>
     <version.org.wildfly.core>6.0.2.Final</version.org.wildfly.core>
-    <version.org.wildfly.common>1.2.0.Final</version.org.wildfly.common>
+    <version.org.wildfly.common>1.5.0.Final-format-001</version.org.wildfly.common>
```
Comment (mareknovotny) [inline pom.xml]: Is that version string correct? `-format-001` is certainly wrong

**Comment by mareknovotny** [comment]
it would be also good to point to WILDFLY/QUARKUS jira if that is the classloading problem and they require a minimal set of dependencies or versions for xnio and wildfly-common

**Comment by ge0ffrey** [comment]
Jenkins execute full downstream build

File: pom.xml (hunk 1 above)
Comment (ge0ffrey) [reply pom.xml]: Fixed

**Comment by ge0ffrey** [comment]
The full downstream build has 2 unrelated failures due to Firefox and Selenium:

```
org.openqa.selenium.ElementClickInterceptedException: 
Element <button id="deploy" class="btn btn-default kie-wb-common-library-project-actions-dropdown-button" type="button"> is not clickable at point (1166.5,128.35000610351562) because another element <div id="panel-id" class="modal in"> obscures it
Build info: version: '3.11.0', revision: 'e59cfb3', time: '2018-03-11T20:26:55.152Z'
System info: host: 'rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com', ip: '127.0.0.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-1062.el7.x86_64', java.version: '1.8.0_202'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Capabilities {acceptInsecureCerts: true, browserName: firefox, browserVersion: 60.2.0, javascriptEnabled: true, moz:accessibilityChecks: false, moz:geckodriverVersion: 0.24.0, moz:headless: true, moz:processID: 27149, moz:profile: /tmp/rust_mozprofile.ZDt5Y4..., moz:useNonSpecCompliantPointerOrigin: false, moz:webdriverClick: true, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, platformVersion: 3.10.0-1062.el7.x86_64, rotatable: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}}
Session ID: 046fe6b0-5102-452e-9101-cb5ef9e7aaa7
	at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.deployAndCheckArtifact(ProjectLibraryIntegrationTest.java:78)
```

**Pull request merged**